packages/askui-nodejs/src/execution/inference-client.ts

```diff
@@ -8,10 +8,22 @@ import { resizeBase64ImageWithSameRatio } from '../utils/transformations';
 import { IsImageRequired } from './is-image-required-interface';
 
 export class InferenceClient {
+  url: string;
+
   constructor(
-    public url: string,
+    public baseUrl: string,
     public httpClient: HttpClientGot,
-  ) { }
+    readonly workspaceId?: string,
+    public apiVersion = 'v2',
+  ) {
+    this.url = workspaceId ? urljoin(
+      this.baseUrl,
+      'api',
+      this.apiVersion,
+      'workspaces',
+      workspaceId,
+    ) : urljoin(this.baseUrl, 'api', this.apiVersion);
```

Comment:
To make this code a little clearer, can we do?
```suggestion
    const versionedBaseUrl = urljoin(this.baseUrl, 'api', this.apiVersion);
    this.url = workspaceId ? urljoin(versionedBaseUrl, 'workspaces', workspaceId) : versionedBaseUrl;
```